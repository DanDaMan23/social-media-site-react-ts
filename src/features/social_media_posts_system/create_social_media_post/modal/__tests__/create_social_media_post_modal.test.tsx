import { render, screen, waitFor, fireEvent } from "@testing-library/react"
import { FC, useEffect, useState } from "react"
import { MockSocialMediaPostsContextProvider } from "../../../contexts/context_test_helpers/social-media-posts.context.test-helpers"
import CreateSocialMedialPostModal from "../create_social_media_post_modal"

interface IRenderCreateModal {
  show?: boolean
  createNewPost?: jest.Mock
  onClose?: jest.Mock
}

const renderCreateModal = ({
  show = true,
  createNewPost = jest.fn(),
  onClose = jest.fn()
}: IRenderCreateModal = {}) => {
  const TestComponent: FC = () => {
    const [showModal, setShowModal] = useState<boolean>(false)

    useEffect(() => {
      setShowModal(show)
    }, [])

    return (
      <MockSocialMediaPostsContextProvider createNewPost={createNewPost}>
        <CreateSocialMedialPostModal show={showModal} onClose={onClose} />
      </MockSocialMediaPostsContextProvider>
    )
  }

  return render(<TestComponent />)
}

describe("create-posts-modal test", () => {
  it("should render correctly when show is true", () => {
    const { baseElement } = renderCreateModal()

    expect(baseElement).toMatchSnapshot()
  })

  it("should not show up when show is false", () => {
    const { baseElement } = renderCreateModal({ show: false })

    expect(baseElement).toMatchSnapshot()
  })

  it("should check when submit button can be clicked or not", async () => {
    renderCreateModal()

    const postContentInput = screen.getByLabelText("Post Content")
    const submitButton = screen.getByText("Submit")

    fireEvent.change(postContentInput, { target: { value: "Hello World" } })
    fireEvent.submit(submitButton)

    await waitFor(() => {
      expect(submitButton).not.toBeDisabled()
    })
  })
})
