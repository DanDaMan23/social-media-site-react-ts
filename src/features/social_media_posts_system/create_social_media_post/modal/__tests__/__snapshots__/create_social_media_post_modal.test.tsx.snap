// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create-posts-modal test should not show up when show is false 1`] = `
<body
  class=""
  style="padding-right: 1024px;"
>
  <div />
</body>
`;

exports[`create-posts-modal test should render correctly when show is true 1`] = `
<body
  class="modal-open"
  data-rr-ui-modal-open=""
  style="padding-right: 1024px;"
>
  <div />
  <div
    class="fade modal-backdrop show"
  />
  <div
    aria-modal="true"
    class="fade modal show"
    role="dialog"
    style="display: block; padding-right: 0px;"
    tabindex="-1"
  >
    <div
      class="modal-dialog"
    >
      <div
        class="modal-content"
      >
        <div
          class="modal-header"
        >
          <div
            class="modal-title h4"
          >
            Create Post
          </div>
          <button
            aria-label="Close"
            class="btn-close"
            type="button"
          />
        </div>
        <div
          class="modal-body"
        >
          <form
            class=""
          >
            <div
              class="mb-3"
            >
              <textarea
                class="form-control"
                id="postContent"
                name="postContent"
                rows="10"
              />
            </div>
          </form>
        </div>
        <div
          class="modal-footer"
        >
          <button
            class="btn btn-secondary"
            type="button"
          >
            Close
          </button>
          <button
            class="btn btn-primary"
            disabled=""
            type="button"
          >
            Submit
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
